backend/handler/igdb_handler.py

```diff
@@ -317,83 +361,164 @@ async def get_rom(self, file_name: str, platform_idgb_id: int) -> IGDBRom:
+            expansions=rom.get("expansions", []),
+            dlcs=rom.get("dlcs", []),
+            remasters=rom.get("remasters", []),
+            remakes=rom.get("remakes", []),
+            expanded_games=rom.get("expanded_games", []),
+            companies=rom.get("involved_companies", []),
+            first_release_date=rom.get("first_release_date", None),
+        )
 
     @check_twitch_token
     def get_matched_roms_by_id(self, igdb_id: int) -> list[IGDBRom]:
         matched_rom = self.get_rom_by_id(igdb_id)
         matched_rom.update(
-            url_cover=matched_rom["url_cover"].replace("t_thumb", "t_cover_big"),
+            url_cover=matched_rom.get("url_cover", "").replace(
+                "t_thumb", "t_cover_big"
+            ),
         )
         return [matched_rom]
 
     @check_twitch_token
     def get_matched_roms_by_name(
-        self, search_term: str, platform_idgb_id: int
+        self, search_term: str, platform_idgb_id: int, search_extended: bool = False
     ) -> list[IGDBRom]:
         if not platform_idgb_id:
             return []
 
+        search_term = uc(search_term)
         matched_roms = self._request(
-            self.games_url,
-            data=f"""
-                search "{uc(search_term)}";
-                fields id, slug, name, summary;
-                where platforms=[{platform_idgb_id}];
-            """,
+            self.games_endpoint,
+            data=f'search "{search_term}"; fields {",".join(self.games_fields)}; where platforms=[{platform_idgb_id}];',
         )
 
+        if not matched_roms or search_extended:
+            log.info("Extended searching...")
+            alternative_matched_roms = self._request(
+                self.search_endpoint,
+                data=f'fields {",".join(self.search_fields)}; where game.platforms=[{platform_idgb_id}] & (name ~ *"{search_term}"* | alternative_name ~ *"{search_term}"*);',
+            )
+
+            if alternative_matched_roms:
+                alternative_roms_ids = []
+                for rom in alternative_matched_roms:
+                    alternative_roms_ids.append(
+                        rom.get("game").get("id", "")
+                        if "game" in rom.keys()
+                        else rom.get("id", "")
+                    )
+                id_filter = " | ".join(
+                    list(
+                        map(
+                            lambda rom: f'id={rom.get("game").get("id", "")}'
+                            if "game" in rom.keys()
+                            else f'id={rom.get("id", "")}',
+                            alternative_matched_roms,
+                        )
+                    )
+                )
+                alternative_matched_roms = self._request(
+                    self.games_endpoint,
+                    f'fields {",".join(self.games_fields)}; where {id_filter};',
+                )
+                matched_roms.extend(alternative_matched_roms)
```

Comment:
this is wild, need to look at it a second time

Reply:
If we get for example two ``alternative_matched_roms`` searching with the ``search`` endpoint, each one with it's own id (lets say 15 and 16), we are building a request like this to the ``games`` endpoint: fields field1, field2; where id=15 | id=16;
After that, we add the result roms to the already matched ones